models/errors_whitebox_test.go

```diff
@@ -0,0 +1,30 @@
+package models
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/almighty/almighty-core/resource"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSimpleError_Error(t *testing.T) {
+	t.Parallel()
+	resource.Require(t, resource.UnitTest)
+	e := simpleError{message: "foo"}
+	assert.Equal(t, "foo", e.Error())
+}
+
+func TestBadParameterError_Error(t *testing.T) {
+	t.Parallel()
+	resource.Require(t, resource.UnitTest)
+	e := BadParameterError{parameter: "foo", value: "bar"}
+	assert.Equal(t, fmt.Sprintf(stBadParameterErrorMsg, e.parameter, e.value), e.Error())
+}
+
+func TestNotFoundError_Error(t *testing.T) {
+	t.Parallel()
+	resource.Require(t, resource.UnitTest)
+	e := NotFoundError{entity: "foo", ID: "bar"}
+	assert.Equal(t, fmt.Sprintf(stNotFoundErrorMsg, e.entity, e.ID), e.Error())
```

Comment:
I understand that part. But what your verifying is the formatting of the error string message. We have no strict requirement around these messages at this time and they are highly subject to rapid change. Adding test code to this part is only asking for test failure on any minuscule code change moving forward. If you wanted to 'touch' the errors to get coverage, a better strategy would be to test the scenarios/paths where they are used. That probably has more meaning.